src/memoize.js

```diff
@@ -0,0 +1,67 @@
+const { isPromise } = require('./utilities')
+
+const stringify = function (...args) {
+  return String(args[0])
+}
+
+module.exports.memoize = function (fn, s = stringify, ms = -1) {
+  if (typeof s === 'number') {
+    ms = s
+    s = stringify
+  }
+
+  const cache = new Map()
+  const timeouts = new Map()
+
+  const append = function (key, value) {
+    if (ms !== -1) {
+      timeouts.set(key, setTimeout(function () {
+        timeouts.delete(key)
+        cache.delete(key)
+      }, ms))
+    }
+
+    cache.set(key, value)
+    return value
+  }
+
+  const m = function (...args) {
+    const key = s(...args)
+    if (cache.has(key)) {
+      return cache.get(key)
```

Comment:
Shouldn't you return `Promise.resolve(cache.get(key))` instead of sometimes returning a promise and sometimes returning a resolved value?  It doesn't make a difference if you're accessing it via `await`, but consistent return otherwise makes using it easier.